### PR TITLE
Codebase: Removed class Disposable

### PR DESCRIPTION
**Pull request by cinqmilleans**
**Resolved / Related Issues**
The abstract class Disposable is very generic. This class is moved to Shared so that it can be used in the SDK.

**Validation**
How did you test these changes?
- [x] Built and ran the app
- [ ] Tested the changes for accessibility

### CODE REVIEW

File: src/Files.Shared/Disposable.cs [hunk 1]
```diff
@@ -1,6 +1,6 @@
 ﻿using System;
 
-namespace Files.App.Shell
+namespace Files.Shared
 {
 	public abstract class Disposable : IDisposable
```
Comment (d2dyno1) [inline src/Files.Shared/Disposable.cs:5]: I think we should get rid of it

Comment (cinqmilleans) [reply src/Files.Shared/Disposable.cs]: It's removed.

File: src/Files.App/Helpers/FileOperationsHelpers.cs [hunk 2]
```diff
@@ -914,7 +914,13 @@ public void WaitForCompletion()
 				operationsCompletedEvent.WaitOne();
 			}
 
-			protected override void Dispose(bool disposing)
+			public void Dispose()
+			{
+				Dispose(true);
+				GC.SuppressFinalize(this);
```
Comment (d2dyno1) [inline src/Files.App/Helpers/FileOperationsHelpers.cs:920]: I'd remove that method and put everyting that's in `Dispose(bool)` here

Comment (yaira2) [reply src/Files.App/Helpers/FileOperationsHelpers.cs:920]: @cinqmilleans

**Pull request closed without merging**